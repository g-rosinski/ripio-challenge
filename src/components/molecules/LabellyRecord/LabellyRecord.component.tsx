import React from "react"
import { SpaceLabelBetweenWrapper } from "./LabellyRecord.styles"
import { DividerHorizontal, Label } from "../../atoms"

type LabellyRecordProps = {
    label: string,
    value: string,
    displayDivider?: boolean
}

export const LabellyRecord: React.FC<LabellyRecordProps> = ({
    label,
    value,
    displayDivider = true
}) => (
    <>
        <SpaceLabelBetweenWrapper>
            <Label>{label}:</Label>
            <Label>{value}</Label>
        </SpaceLabelBetweenWrapper>
        {displayDivider && <DividerHorizontal role="presentation" />}
    </>
)