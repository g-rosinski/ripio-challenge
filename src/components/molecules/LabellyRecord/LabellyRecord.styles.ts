import styled from "styled-components";

export const SpaceLabelBetweenWrapper = styled.div`
    display: flex;
    & > label:last-child{
        text-align: right;
        font-weight: 700;
    }
`