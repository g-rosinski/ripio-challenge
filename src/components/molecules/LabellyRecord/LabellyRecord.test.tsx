import { render, screen } from "@testing-library/react"
import React from "react"
import { LabellyRecord } from "./LabellyRecord.component"

describe('<LabellyRecord />', () => {

    const baseProps = { 
        label: 'Cuenta de destino', 
        value: 'receptor@mail.com',
    }

    test('Se reenderice Correctamente', () => {
        const { container } = render(<LabellyRecord {...baseProps} />)

        expect(screen.getByRole('presentation')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })

    test('Que no se muestre el divider si displayDivider=false', () => {
        render(<LabellyRecord {...baseProps} displayDivider={false} />)

        expect(screen.queryByRole('presentation')).toBeFalsy()
    })
})