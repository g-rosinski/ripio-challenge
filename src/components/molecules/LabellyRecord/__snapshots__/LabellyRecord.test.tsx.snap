// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabellyRecord /> Se reenderice Correctamente 1`] = `
<div>
  <div
    class="sc-bcXHqe iqcvMI"
  >
    <label
      class="sc-hLBbgP ksCyWo"
    >
      Cuenta de destino
      :
    </label>
    <label
      class="sc-hLBbgP ksCyWo"
    >
      receptor@mail.com
    </label>
  </div>
  <hr
    class="sc-pyfCe dkPJOj"
    role="presentation"
  />
</div>
`;
